vehicle/mkvi/software/imu/imu.c

```diff
@@ -0,0 +1,168 @@
+#include "imu_config.h"
+#include "libs/gpio/api.h"
+#include "libs/gpio/pin_defs.h"
+#include "libs/timer/api.h"
+#include "vehicle/mkvi/software/imu/can_api.h"
+#include "vehicle/mkvi/software/imu/icm20948/icm20948.h"
+
+#include <avr/interrupt.h>
+#include <avr/io.h>
+#include <stdint.h>
+#include <util/delay.h>
+
+// #include "projects/btldr/btldr_lib.h"
+// #include "projects/btldr/git_sha.h"
+// #include "projects/btldr/libs/image/api.h"
+// #include "projects/can_print/can_print.h"
+
+// /*
+//  * Required for btldr
+//  */
+// image_hdr_t image_hdr __attribute__((section(".image_hdr"))) = {
+//     .image_magic = IMAGE_MAGIC,
+//     .git_sha = STABLE_GIT_COMMIT,
+// };
+
+/**
+ * Timer 0 for sending IMU data over CAN
+ */
+volatile bool can_send_imu_data = true;
+void timer_0_isr(void) {
+    can_send_imu_data = true;
+}
+
+/**
+ * Timer 1 for the LED heartbeat
+ */
+volatile bool led_heartbeat = true;
+void timer_1_isr(void) {
+    led_heartbeat = true;
+}
+
+uint8_t test = 0;
+/**
+ * Update the arb id of an IMU based on soldered pull up jumpers.
+ *
+ * Allows for multiple IMUs running the same firmware on the same CAN bus.
+ */
+void update_can_arb_id(void) {
```

Comment:
not in the scope of this PR, but in the future, we should expand the functionality of our CAN API to handle boards with multiple can IDs for the same message - that way, when our DBC gets assembled, we can automatically detect if there's id collisions